### ISSUE TITLE
Accessing CuDNN autotuner in built-in Keras

### ISSUE THREAD

**Issue by ASvyatkovskiy** [tensorflow/tensorflow#18913]
### System information
- **Have I written custom code (as opposed to using a stock example script provided in TensorFlow)**: No
- **OS Platform and Distribution (e.g., Linux Ubuntu 16.04)**:
```bash
$ uname -mrs
Linux 4.14.0-49.el7a.ppc64le ppc64le
```
- **TensorFlow installed from (source or binary)**: Binary
- **TensorFlow version (use command below)**: 1.6.0
- **Python version**: 2.7.5
- **Bazel version (if compiling from source)**:
- **GCC/Compiler version (if compiling from source)**:
- **CUDA/cuDNN version**: CUDA 9.1, CuDNN 7.0.5
- **GPU model and memory**: Tesla V100:
```
== nvidia-smi ===================================================
Thu Apr 26 18:35:34 2018       
+-----------------------------------------------------------------------------+
| NVIDIA-SMI 396.19                 Driver Version: 396.19                    |
|-------------------------------+----------------------+----------------------+
| GPU  Name        Persistence-M| Bus-Id        Disp.A | Volatile Uncorr. ECC |
| Fan  Temp  Perf  Pwr:Usage/Cap|         Memory-Usage | GPU-Util  Compute M. |
|===============================+======================+======================|
|   0  Tesla V100-SXM2...  Off  | 00000004:04:00.0 Off |                    0 |
| N/A   29C    P0    52W / 300W |      0MiB / 15360MiB |      0%   E. Process |
+-------------------------------+----------------------+----------------------+
```
- **Exact command to reproduce**: N/A

We would like to be able to have access to the CuDNN autotuner in `tf.keras` module to access optimal algorithms for a given hardware (or, perhaps passing a custom convolutional algorithm from config). In TensorFlow, I can specify to use the CuDNN autotuner by setting: 
```bash
os.environ['TF_CUDNN_USE_AUTOTUNE'] = "1"
```
(currently enabled by default), which improves performance significantly on Volta GPUs and especially with FP16.

However, I am unable to access this performance improvement when running pure `tf.keras`, where setting this environmental variable does not have any effect. 

### Source code / logs

Following simple example could be used to reproduce the issue: https://gist.github.com/ASvyatkovskiy/8d1dd622e447d9d8de1ec4e238e0dbaa

**Comment by tensorflowbutler** [comment]
Thank you for your post. We noticed you have not filled out the following field in the issue template. Could you update them if they are relevant in your case, or leave them as N/A? Thanks.
Exact command to reproduce

**Comment by tensorflowbutler** [comment]
It has been 14 days with no activity and the `awaiting response` label was assigned. Is this still an issue?

**Comment by bhack** [comment]
Check also https://github.com/keras-team/keras/issues/9825 /cc @fchollet

**Comment by bhack** [comment]
[Off-topic] @fchollet I note that more and more issues are a little bit duplicated after keras gone in the tf namespace. I think it is quite natural that people are a little bit cofused on where to report issues. I think that TF is a special case backend that it is also a sort of frontend in tf.keras. Of course keras is the upstream but this frontend/backend/frontend loop it is confusing many users.

**Comment by tensorflowbutler** [comment]
It has been 18 days with no activity and the `awaiting response` label was assigned. Is this still an issue?

**Comment by tensorflowbutler** [comment]
It has been 33 days with no activity and the `awaiting response` label was assigned. Is this still an issue?

**Comment by tensorflowbutler** [comment]
We are closing this issue for now due to lack of activity. Please comment if this is still an issue for you. Thanks!

**Comment by bhack** [comment]
Please reopen.
